### PR TITLE
Add Pose Estimation Tensorflow Processor

### LINKED ISSUE #15: Add Real-time human pose estimation processor
The real-time multi-person pose estimation is a computer vision techniques that detect human poses in images and video.  Recent advancement in DL architectures has provided a novel, faster and more accurate solutions for the multi-person pose estimation task.  The most prominent articles and related implementations include: 
- [Realtime Multi-Person 2D Pose Estimation using Part Affinity Fields](https://arxiv.org/pdf/1611.08050.pdf)
  - [OpenPose](https://github.com/ildoonet/tf-pose-estimation)
  - [tf-pose-estimation](https://github.com/ildoonet/tf-pose-estimation)
- [PersonLab: Person Pose Estimation and Instance Segmentation with a Bottom-Up, Part-Based, Geometric Embedding Model](https://arxiv.org/pdf/1803.08225.pdf)
  - [PoseNet](https://github.com/tensorflow/tfjs-models/tree/master/posenet)

Adopt some of the above techniques/models to implement a SCDF processor for real-time multi-person pose estimation.

### PR DESCRIPTION
**Pull request by tzolov**
Resolves #15

     - Add new pose-estimation tensorflow sub-project
     - Implementation is based on openpose paper and related openpose and tf-pose-estimation projects.
     - Requires pre-trained Tensorflow models build with the tf-pose-estimation project.
     - Post-processing, implemented in Java!
     - Add integration tests
     - Add README docs

### CODE REVIEW

File: spring-cloud-starter-stream-processor-pose-estimation/src/test/java/org/springframework/cloud/stream/app/pose/estimation/processor/PoseEstimationTensorflowProcessorIntegrationTests.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+				JSONAssert.assertEquals(expected, new JSONArray(received.getPayload()), false);
+			}
+		}
+
+		@TestPropertySource(properties = {
+				"tensorflow.mode=header",
+				//"tensorflow.model=file:/Users/ctzolov/Dev/projects/tf-pose-estimation/models/graph/mobilenet_thin/graph_opt.pb",
+				"tensorflow.model=file:/Users/ctzolov/Dev/projects/tf-pose-estimation/models/graph/cmu/graph_opt.pb",
```
Comment (ilayaperumalg) [inline spring-cloud-starter-stream-processor-pose-estimation/src/test/java/org/springframework/cloud/stream/app/pose/estimation/processor/PoseEstimationTensorflowProcessorIntegrationTests.java]: @tzolov Can we replace this with some local file reference?

Comment (tzolov) [reply spring-cloud-starter-stream-processor-pose-estimation/src/test/java/org/springframework/cloud/stream/app/pose/estimation/processor/PoseEstimationTensorflowProcessorIntegrationTests.java]: Good point. 
I will fix this for the other processors tests as well

**Comment by ilayaperumalg** [comment]
LGTM, merging.

**Comment by ilayaperumalg** [comment]
Squashed and merged as 5e9c5a1

**Pull request closed without merging**